dl1_data_handler/reader.py

```diff
@@ -961,6 +961,53 @@ def _transform_to_sky_spher_offsets(self, table) -> Table:
         table.add_column(angular_separation, name="angular_separation")
         return table
 
+    def get_parameters_dict(self, batch, parameter_list) -> Dict:
+        """
+        Retrieve a dictionary of existing DL1b parameters for a given batch.
+
+        Parameters
+        ----------
+        batch : astropy.table.Table
+            A Table containing the batch with columns `file_index`, `table_index`, and `tel_id`.
+        parameter_list : list
+            List of DL1b parameters to retrieve.
+
+        Returns
+        -------
+        param_dict : dict
+            Dictionary where keys are parameter names and values are lists of parameter values across the batch.
+        """
+        param_dict = {param: [] for param in parameter_list}
+        initialized = False
+        available_params = set()
+
+        for file_idx, table_idx, tel_id in batch.iterrows(
+            "file_index", "table_index", "tel_id"
+        ):
+            filename = list(self.files)[file_idx]
+            tel_table = f"tel_{tel_id:03d}"
+
+            with lock:
+                child = self.files[
+                    filename
+                ].root.dl1.event.telescope.parameters._f_get_child(tel_table)
+
+                if not initialized:
+                    available_params = set(child.dtype.names)
+                    param_dict = {
+                        param: []
+                        for param in parameter_list
+                        if param in available_params
+                    }
+                    initialized = True
+
+                row = child[table_idx]
+
+                for param in param_dict:
+                    param_dict[param].append(row[param])
+
+        return param_dict
+
     def get_parameters(self, batch, dl1b_parameter_list) -> np.array:
```

Comment:
Maybe also please remove the argument `dl1b_parameter_list` and just return all dl1b parameter for the given batch. You have all the column names here: `self.dl1b_parameter_colnames`